python/semantic_kernel/connectors/memory/azure_cosmosdb/azure_cosmos_db_memory_store.py

```diff
@@ -0,0 +1,470 @@
+# Copyright (c) Microsoft. All rights reserved.
+import json
+from abc import ABC, abstractmethod
+from typing import List, Tuple
+
+import numpy as np
+from numpy import ndarray
+from pymongo.mongo_client import MongoClient
+
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+from semantic_kernel.utils.settings import azure_cosmos_db_settings_from_dot_env
+
+# Load environment variables
+(
+    cosmos_api,
+    cosmos_connstr,
+    cosmos_database_name,
+    cosmos_container_name,
+) = azure_cosmos_db_settings_from_dot_env()
+
+# OpenAI Ada Embeddings Dimension
+VECTOR_DIMENSION = 1536
+
+
+# Abstract class similar to the original data store that allows API level abstraction
+class AzureCosmosDBStoreApi(ABC):
+    @abstractmethod
+    async def ensure(self, num_lists, similarity, vector_dimensions):
+        raise NotImplementedError
+
+    @abstractmethod
+    async def create_collection_core(self, collection_name: str) -> None:
```

Comment:
Regarding methods with `_core` ending in name - I'm not sure it's required. We can keep it simple and name it `create_collection`, `get_collections`, `delete_collection` etc.

Reply:
Renamed the methods.